lib/rubocop/cop/rspec/class_check.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Enforces consistent use of `be_a` or `be_kind_of`.
+      #
+      # @example EnforcedStyle: be_a (default)
+      #   # bad
+      #   expect(var).to be_kind_of(Date)
+      #   expect(var).to be_kind_of(Integer)
+      #
+      #   # good
+      #   expect(var).to be_a(Date)
+      #   expect(var).to be_a(Integer)
+      #
+      # @example EnforcedStyle: be_kind_of
+      #   # bad
+      #   expect(var).to be_a(Time)
+      #   expect(var).to be_a(String)
+      #
+      #   # good
+      #   expect(var).to be_kind_of(Time)
+      #   expect(var).to be_kind_of(String)
+      #
+      class ClassCheck < Base
+        extend AutoCorrector
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Prefer `%<preferred>s` over `%<current>s`.'
+
+        RESTRICT_ON_SEND = %i[
+          be_a
```

Comment:
there are also `be_an` and `be_a_kind_of`. While be_a / be_an is being used depending on the argument (e.g. they are the same style), it would be nice in the case of be_kind_of style both to be corrected; and also to support the be_a_kind_of  style.

Reply:
Good catch. I'll support them too 👍